watsong/jukebox.py

```diff
@@ -27,15 +35,18 @@ def jukebox() -> Any:
 
         if query:
             album_descs, err = watson.get_albums(query)
+
             if err is not None:
                 flash(str(err))
                 return render_template("jukebox.html", songs=songs, dials=DIALS)
-            spotify.cache(album_descs)
 
-            songs = spotify.get_songs(album_descs)
+            if not current_app.testing:
+                spotify.cache(album_descs, current_app.spotify)
```

Comment:
If we update the cache while testing, everything gets set to "What's a girl to do"